### PR TITLE
[rag eval] fix crashing question, add loading, fix answer showing instead of questions

### PR DESCRIPTION
**Pull request by chejennifer**
- fix crashing when a query doesn't have an answer
- add loading when fetching data for the feedback section
- update reading either answer or questions in the LHS

### CODE REVIEW

File: static/js/apps/eval_retrieval_generation/call_feedback.tsx [hunk 1]
```diff
@@ -103,19 +104,29 @@ export function CallFeedback(): JSX.Element {
...
+          setEvalInfo({
+            dcResponse: `${row.get(DC_RESPONSE_COL)}${tableResponse}`,
+            dcStat: row.get(DC_STAT_COL),
+            llmStat: row.get(LLM_STAT_COL),
+            question: row.get(DC_QUESTION_COL),
+          });
+        }
+      })
+      .catch((e) => {
+        console.log(e);
```
Comment (shifucun) [inline static/js/apps/eval_retrieval_generation/call_feedback.tsx:124]: alert here?

File: static/js/apps/eval_retrieval_generation/feedback_navigation.tsx [hunk 2]
```diff
@@ -156,8 +156,8 @@ export function FeedbackNavigation(
       while (!(targetId in userQuery)) {
         targetId -= 1;
       }
-      setSessionQueryId(targetId);
       setFeedbackStage(getFirstFeedbackStage(evalType));
+      setSessionQueryId(targetId);
```
Comment (shifucun) [inline static/js/apps/eval_retrieval_generation/feedback_navigation.tsx:160]: why the order here matters?

Comment (chejennifer) [reply static/js/apps/eval_retrieval_generation/feedback_navigation.tsx:160]: I think it has to do with the pane listening on both feedback stage and session query id changes and I'm not sure it'll be better, but just switched it to the same order as eval list choosing a new query

File: static/js/apps/eval_retrieval_generation/call_feedback.tsx (hunk 1 above)
Comment (chejennifer) [reply static/js/apps/eval_retrieval_generation/call_feedback.tsx]: done

File: static/js/apps/eval_retrieval_generation/feedback_navigation.tsx (hunk 2 above)
Comment (shifucun) [reply static/js/apps/eval_retrieval_generation/feedback_navigation.tsx:160]: i think react setState is applied async and in batch, so the order here should not matter.

**Review by shifucun — APPROVED** [review]
thanks!

**Comment by chejennifer** [comment]
thanks for the review!

**Pull request merged**